src/input/controller.js

```diff
@@ -125,6 +125,20 @@ class Controller {
         }
     }
 
+    /**
+     * Reusable function to check and add actions
+     *
+     * @param {string} action - The name of the action.
+     * @param {object} item - An object to add.
+     */
+    appendAction(action, item) {
+        if (this._actions[action]) {
+            this._actions[action].push(item);
+        } else {
+            this._actions[action] = [item];
+        }
```

Comment:
```suggestion
        this._actions[action] = this._actions[action] || [];
        this._actions[action].push(item);
```

(IMO this creates less confusion and less potential bugs like the +- issue)